util/log/file.go

```diff
@@ -85,36 +97,92 @@ func shortHostname(hostname string) string {
 	return hostname
 }
 
-// logName returns a new log file name containing tag, with start time t, and
-// the name for the symlink for tag.
-func logName(tag string, t time.Time) (name, link string) {
-	name = fmt.Sprintf("%s.%s.%s.log.%s.%04d%02d%02d-%02d%02d%02d.%d",
-		program,
-		host,
-		userName,
-		tag,
-		t.Year(),
-		t.Month(),
-		t.Day(),
-		t.Hour(),
-		t.Minute(),
-		t.Second(),
+// escapeString replaces all the periods in the string with an underscore, and
+// every underscore with a double underscore.
+func escapeStringForFilename(s string) string {
```

Comment:
I'm assuming this is only used for a fixed format, so maybe easier to just deal with that format rather than trying to have something general which needs to think about all sorts of situations irrelevant to you.